examples/custom-client-cxx/cli.cpp

```diff
@@ -0,0 +1,150 @@
+  }
+
+  std::unique_ptr<AkliteClient> client;
+  try {
+    client = std::make_unique<AkliteClient>(cfg_dirs);
+  } catch (const std::exception& exc) {
+    LOG_ERROR << "Failed to initialize the client: " << exc.what();
+    return EXIT_FAILURE;
+  }
+
+  auto ret = aklite::cli::CheckIn(*client, is_local_update ? &local_update_source : nullptr);
+  return aklite::cli::IsSuccessCode(ret)? EXIT_SUCCESS : EXIT_FAILURE;
+}
+
+int cmd_pull(std::string target_name, std::string local_repo_path) {
+  boost::log::core::get()->set_filter(boost::log::trivial::severity >= boost::log::trivial::info);
+  bool is_local_update{false};
+  LocalUpdateSource local_update_source;
+
+  std::vector<boost::filesystem::path> cfg_dirs;
+  auto env{boost::this_process::environment()};
+  if (env.end() != env.find("AKLITE_CONFIG_DIR")) {
+    cfg_dirs.emplace_back(env.get("AKLITE_CONFIG_DIR"));
+  } else {
+    cfg_dirs = AkliteClient::CONFIG_DIRS;
+  }
+
+  if (!local_repo_path.empty()) {
+    local_update_source = {
+      local_repo_path + "/tuf",
+      local_repo_path + "/ostree_repo",
+      local_repo_path + "/apps"
+    };
+    is_local_update = true;
+    LOG_INFO << "Offline mode. Updates path=" << local_repo_path;
+  } else {
+    LOG_INFO << "Online mode";
+  }
+
+  std::unique_ptr<AkliteClient> client;
+  try {
+    client = std::make_unique<AkliteClient>(cfg_dirs);
+  } catch (const std::exception& exc) {
+    LOG_ERROR << "Failed to initialize the client: " << exc.what();
+    return EXIT_FAILURE;
+  }
+
+  auto ret = aklite::cli::Pull(*client, -1, "", InstallMode::All, true, is_local_update ? &local_update_source : nullptr);
+  return aklite::cli::IsSuccessCode(ret)? EXIT_SUCCESS : EXIT_FAILURE;
+}
+
+int cmd_install(std::string target_name, std::string local_repo_path) {
+  boost::log::core::get()->set_filter(boost::log::trivial::severity >= boost::log::trivial::info);
+  bool is_local_update{false};
+  LocalUpdateSource local_update_source;
+
+  std::vector<boost::filesystem::path> cfg_dirs;
+  auto env{boost::this_process::environment()};
+  if (env.end() != env.find("AKLITE_CONFIG_DIR")) {
+    cfg_dirs.emplace_back(env.get("AKLITE_CONFIG_DIR"));
+  } else {
+    cfg_dirs = AkliteClient::CONFIG_DIRS;
+  }
+
+  if (!local_repo_path.empty()) {
+    local_update_source = {
+      local_repo_path + "/tuf",
+      local_repo_path + "/ostree_repo",
+      local_repo_path + "/apps"
+    };
+    is_local_update = true;
+    LOG_INFO << "Offline mode. Updates path=" << local_repo_path;
+  } else {
+    LOG_INFO << "Online mode";
+  }
+
+  std::unique_ptr<AkliteClient> client;
+  try {
+    client = std::make_unique<AkliteClient>(cfg_dirs);
+  } catch (const std::exception& exc) {
+    LOG_ERROR << "Failed to initialize the client: " << exc.what();
+    return EXIT_FAILURE;
+  }
+
+  auto ret = aklite::cli::Install(*client, -1, "", InstallMode::All, true, is_local_update ? &local_update_source : nullptr, aklite::cli::PullMode::SkipPull);
```

Comment:
Maybe it makes sense to add `switch (ret) case <>:` statement and print message instructing user what to do next depending on the `ret` value.